go/tools/gopackagesdriver/aspect.bzl

```diff
@@ -0,0 +1,116 @@
+load(
+    "//go/private:providers.bzl",
+    "GoArchive",
+    "GoStdLib",
+)
+load(
+    "//go/private:context.bzl",
+    "go_context",
+)
+load(
+    "@bazel_skylib//lib:paths.bzl",
+    "paths",
+)
+load(
+    "@bazel_skylib//lib:collections.bzl",
+    "collections",
+)
+
+GoPkgInfo = provider()
+
+def _is_file_external(f):
+    return f.owner.workspace_root != ""
+
+def _file_path(f):
+    if f.is_source and not _is_file_external(f):
+        return paths.join("__BAZEL_WORKSPACE__", f.path)
+    return paths.join("__BAZEL_EXECROOT__", f.path)
+
+def _go_pkg_info_aspect_impl(target, ctx):
+    # Fetch the stdlib JSON file from the inner most target
+    stdlib_json = None
+
+    deps_transitive_json = []
+    deps_transitive_x = []
+    if hasattr(ctx.rule.attr, "deps"):
+        for dep in ctx.rule.attr.deps:
+            if GoPkgInfo in dep:
+                pkg_info = dep[GoPkgInfo]
+                deps_transitive_json.append(pkg_info.transitive_json)
+                deps_transitive_x.append(pkg_info.transitive_x)
+                # Fetch the stdlib json from the first dependency
+                if not stdlib_json:
+                    stdlib_json = pkg_info.stdlib_json
+
+    # If deps are embedded, do not gather their json or x since they are
+    # included in the current target, but do gather their deps'.
+    if hasattr(ctx.rule.attr, "embed"):
+        for dep in ctx.rule.attr.embed:
+            if GoPkgInfo in dep:
+                pkg_info = dep[GoPkgInfo]
+                deps_transitive_json.append(pkg_info.deps_transitive_json)
+                deps_transitive_x.append(pkg_info.deps_transitive_x)
+
+    pkg_json = None
+    x = None
+    if GoArchive in target:
```

Comment:
FYI, we'll eventually need an action that writes JSON files with more data (AST, type info). I believe gopls only requests the minimum from go/packages, then loads the rest on its own. Other clients will want a more complete response.

Definitely no need to do that now, but this might be a good spot for a TODO.

Reply:
Actually, the `GOPACKAGESDRIVER` should return this type as a result, which is pretty barebone: https://github.com/golang/tools/blob/master/go/packages/packages.go#L417-L433

```go
```